app/app/route.js

```diff
@@ -5,11 +5,16 @@ export default Ember.Route.extend({
     if (tokens.length === 0) {
       tokens.push(this.currentModel.get('handle'));
     }
-    var stack = this.currentModel.get('stack');
+
+    let stack = this.get('stack');
+    Ember.assert('App Route must have a stack', stack);
     tokens.push(stack.get('handle'));
     return tokens.join(' - ');
   },
   afterModel: function(model){
-    return model.get('stack');
+    let route = this;
+    return model.get('stack').then(function(stack){
+      route.set('stack', stack);
```

Comment:
In `title:` you should be able to presume that `stack` has been loaded.

Where did `let stack = this.currentModel.get('stack')` fall down?